src/block_cache/block_mgr.c

```diff
@@ -104,6 +104,26 @@ __bm_checkpoint(
 
     WT_RET(__wt_block_checkpoint(session, block, buf, ckptbase, data_checksum));
 
+    /*
+     * We might have postponed a tiered switch to a new file. This is the right time to make that
+     * happen since eviction is disabled at the moment, and hence we are the exclusive writers.
+     */
+    if (bm->next_block != NULL) {
+        WT_ASSERT(session, bm->switching);
+        __wt_writelock(session, &bm->handle_array_lock);
+        bm->prev_block = bm->block;
+        bm->block = bm->next_block;
+        __wt_writeunlock(session, &bm->handle_array_lock);
+        bm->switching = false;
+    }
+
+    /*
+     * QUESTIONS: If after this point checkpoint fails - do we switch back to older file? Is the
+     * above block better be after the multi handle check, or possibly before the first switch we
+     * are not multi handle just yet? If we are multi handle, and since all the files marked for
+     * sync are fsync here, does that mean we might fsync the prev_block twice now?
+     */
+
```

Comment:
Check if checkpoint can fail, and we don't panic and keep continuing. Maybe, make checkpoints fatal for tiered storage.